### ISSUE TITLE
[HOLD for test] Tooltip appears sometimes when go back from chat (mWeb - Safari) - reported by @aneequeahmad

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#9737]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/docs/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to chat with someone
2. Go back from chat using back button and open chat again and go back.
3. Repeat this process sometimes and a tool tip will appear when back on chat screen

## Expected Result:
No tootip should appear on chats list screen.

## Actual Result:
Tooltip appear on chat list screen

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Mobile Web (Safari)/ iphone 11 pro

**Version Number:** 1.1.79-0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation  

![Image from iOS (13)](https://user-images.githubusercontent.com/43995119/177476052-dcd8d1ee-aff9-461d-a056-505e716693ef.jpg)

https://user-images.githubusercontent.com/43995119/177476054-4f569df4-ac33-4210-a9bd-b0b14a38168b.mov

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @aneequeahmad
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1655759723252049

**Job Post:** https://www.upwork.com/jobs/~01d41ef083193aa30f

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by roryabraham** [comment]
@kevinksullivan as soon as this is exported please make it a monthly

**Comment by metehanozyurt** [comment]
## Proposal
### Cause

When button press tooltip showed up mWeb. 

### Solution

We can restrict tooltip views on mobile webs. We need to get help from the canUseTouchscreen function in our library and make the following changes. In this way, we prevent other tooltips from appearing. While doing the tests, I found that other tooltips were also appearing. We do not encounter this problem only with the back button.

https://github.com/Expensify/App/blob/a2f8cf96d095ded70a15d04ee69bd37be2638c14/src/components/Tooltip/index.js#L142-L148

Adding this line 

```js
        // Skip the tooltip and return the children, if the text is empty.
        if (_.isEmpty(this.props.text)) {
            return this.props.children;
        }
        
        // Skip the tooltip and return the children, if the mweb.
        if (this.canUseTouchScreen) {
            return this.props.children;
        }
        let child = (
```

and this change on constructor

```js
....
        this.canUseTouchScreen = canUseTouchScreen();
        this.state = {
            // Is tooltip rendered?
....
```

**Comment by Santhosh-Sellavel** [comment]
I'm unable to reproduce this issue itself.

**Comment by kevinksullivan** [comment]
Price doubled

**Comment by Santhosh-Sellavel** [comment]
Proposals seems to be redundant, we don't have a consistent reproduction steps either.

**Comment by aneequeahmad** [comment]
@Santhosh-Sellavel there are consistent reproduction steps.

**Action performed :**
- Open any report in mWeb.
- Long press back button in report screen and see the when navigated back `back tooltip is shown`

**Video attached:**

https://user-images.githubusercontent.com/36359331/187566049-699a2af8-e73e-492f-8f47-f7420f8e4e9b.mov


Could i ask why is the proposal redundant ? this code `matchMedia()` is very reliable way to check` if the primary input mechanism is a pointing device.`  detail can be seen [here](https://www.w3schools.com/jsref/api_mediaquerylist.asp#:~:text=page%20out%20loud-,Media%20Features,-Value).

Also we can use following MediaQueryList Object features 

- [any-hover](https://www.w3schools.com/jsref/api_mediaquerylist.asp#:~:text=Does%20any%20available%20input%20mechanism%20allow%20the%20user%20to%20hover%20over%20elements%3F%20(added%20in%20Media%20Queries%20Level%204))  => Does any available input mechanism allow the user to hover over elements? (added in Media Queries Level 4)

- [any-pointer](https://www.w3schools.com/jsref/api_mediaquerylist.asp#:~:text=Is%20any%20available%20input%20mechanism%20a%20pointing%20device%2C%20and%20if%20so%2C%20how%20accurate%20is%20it%3F%20(added%20in%20Media%20Queries%20Level%204)) => Is any available input mechanism a pointing device, and if so, how accurate is it? (added in Media Queries Level 4)

**Comment by Santhosh-Sellavel** [comment]
@aneequeahmad No luck for me
Unable to reproduce!

cc: @roryabraham

**Comment by aneequeahmad** [comment]
@Santhosh-Sellavel, i just reproduced this in production https://new.expensify.com/

![IMG_8067](https://user-images.githubusercontent.com/36359331/187717975-d366a69c-1257-4094-b593-8a6758462e68.PNG)

https://user-images.githubusercontent.com/36359331/187717888-0bae9ddc-11ef-4d59-ba19-acd42f849467.MP4

**Comment by Santhosh-Sellavel** [comment]
Try at staging, please!

**Comment by aneequeahmad** [comment]
@Santhosh-Sellavel  I can reproduce on staging as well. 

https://user-images.githubusercontent.com/36359331/187725017-19a39bb6-f9a8-4f8d-b54e-85065f98c5de.MP4

**Comment by Santhosh-Sellavel** [comment]
@roryabraham
I agree the proposal is improvised, thanks for the clarification!

**Comment by Santhosh-Sellavel** [comment]
Any update on the PR @aneequeahmad?

**Comment by aneequeahmad** [comment]
@Santhosh-Sellavel I'm working on unit test coverage and simple render test. I'm not an expert in writing unit tests. Will be hopefully done with it over the weekend. Are there any tests for tooltips already written for reference ? Will update soon here when the PR is ready. Thanks

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by Santhosh-Sellavel** [comment]
PR is up, will review it!

**Comment by roryabraham** [comment]
Left some advice in the PR to help @aneequeahmad get it cleaned up and ready for more reviews

**Comment by kevinksullivan** [comment]
PR in review

**Comment by roryabraham** [comment]
We've given @aneequeahmad until tomorrow to provide a PR in a mergeable state, or we're going to cancel the job as a failure, at which point I'll most likely just take over the PR myself.

**Comment by aneequeahmad** [comment]
@roryabraham You're right this is taking too long to ship. IMO i'm not eligible for this $250 [bonus](https://github.com/Expensify/App/issues/9737#:~:text=I%20would%20be%20willing%20to%20authorize%20a%20%24250%20bonus%20for%20unit%20test%20coverage%20to%20determine%20when%20we%20do%20or%20do%20not%20show%20tooltips%2C%20and%20a%20simple%20render%20test%20to%20determine%20that%20they%20are%20showing%20when%20we%20want%20them%20to%20and%20not%20showing%20when%20we%20don%27t%20want%20them%20to). You can take over the PR. Shipping the fix ASAP is our priority.

**Comment by kevinksullivan** [comment]
@aneequeahmad can you apply for this for reporting?

https://www.upwork.com/jobs/~0182862270dadf28ac

Going to unassign everyone but @roryabraham and make this internal.

**Comment by aneequeahmad** [comment]
@kevinksullivan Applied, thanks

**Comment by kevinksullivan** [comment]
@roryabraham should we pass this one back to the pool or find someone else to pick it up, given sequence numbers is a higher priority that you will continue to be working on?

**Comment by aneequeahmad** [comment]
@roryabraham, i know you have much pressing priorities and the remaining `test case work` of this PR isn't worth your time. 

Please allow me to work on this. I will not take the compensation for bonus or even if you think i'm not eligible for the `payment of bug fix` i'll accept that too. Will do the requested changes in this week.

Again apologies for dragging this PR to this point.

cc: @kevinksullivan

**Comment by kevinksullivan** [comment]
Discussing next steps on this [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1668640090247899)

**Comment by roryabraham** [comment]
Making some fair progress on https://github.com/Expensify/App/pull/12168 – putting in a bit of extra work make sure our changes are properly covered by tests.

**Comment by roryabraham** [comment]
No update, my PR will probably be on HOLD for a minute for [this PR](https://github.com/Expensify/App/pull/12056) to be merged.

Then I can:

1. Migrate tooltips from web-only [`ReactDOM.createPortal`](https://github.com/Expensify/App/blob/334ea5782120159cb268b59deed540c6c6372033/src/components/Tooltip/TooltipRenderedOnPageBody.js#L125) to the platform-agnostic equivalent being added in https://github.com/Expensify/App/pull/12056
1. Then I can continue my PR and fix the automated tests.

**Comment by roryabraham** [comment]
Still on HOLD for now

**Comment by JmillsExpensify** [comment]
Still working through the held PR, which has additional scope and incomplete regression testing.

**Comment by roryabraham** [comment]
The PR I was waiting for was merged so I can take this off hold now

**Comment by JmillsExpensify** [comment]
Woo! Let's get after it.

**Comment by JmillsExpensify** [comment]
Assigned back @Santhosh-Sellavel. @roryabraham is now unblocked and the draft PR [is here](https://github.com/Expensify/App/pull/12168).

**Comment by Santhosh-Sellavel** [comment]
Thanks!

**Comment by Santhosh-Sellavel** [comment]
@roryabraham Let me know when PR is ready for review!

**Comment by aneequeahmad** [comment]
@JmillsExpensify, I'm also eligible for $500 for the fixing the [issue](https://github.com/Expensify/App/issues/9737#issuecomment-1291982874). It is bonus work of $250 which i'm not eligible of as stated [here](https://github.com/Expensify/App/issues/9737#:~:text=I%20would%20be%20willing%20to%20authorize%20a%20%24250%20bonus%20for%20unit%20test%20coverage%20to%20determine%20when%20we%20do%20or%20do%20not%20show%20tooltips%2C%20and%20a%20simple%20render%20test%20to%20determine%20that%20they%20are%20showing%20when%20we%20want%20them%20to%20and%20not%20showing%20when%20we%20don%27t%20want%20them%20to.). I reported the issue which wasn't easily reproducible and also proposed the solution which was accepted. 

cc: @Santhosh-Sellavel, @roryabraham

**Comment by JmillsExpensify** [comment]
@aneequeahmad I'm not sure I agree with that last comment. Can you help me understand where you fixed the issue. As noted in the linked comment, you were not able to merge the PR so this had to be taken internal.

**Comment by aneequeahmad** [comment]
@JmillsExpensify My accepted [proposal](https://github.com/Expensify/App/issues/9737#:~:text=//%20Skip%20the%20tooltip%20and%20return%20the%20children%2C%20if%20the%20text%20is%20empty%20or%20there%20is%20no%20mouse/pointer/cursor.%0A%20%20%20if%20(_.isEmpty(this.props.text)%20%7C%7C%20!matchMedia(%27(pointer%3Afine)%27).matches)%20%7B%0A%20%20%20%20%20%20%20return%20this.props.children%3B%0A%20%20%20%7D) fixes the bug that was reported(`i reported the bug`). 

According to the [comment](https://github.com/Expensify/App/issues/9737#:~:text=All%20that%20said%2C%20I,don%27t%20want%20them%20to) of reviewer it was a bonus work of $250 which was unit test coverage. 

As it was taking long almost 1.5 months since i was assigned the issue and `the bonus work`(tests) was incomplete but the bug fix was there in the code and i willingly said [here](https://github.com/Expensify/App/issues/9737#:~:text=IMO%20i%27m%20not%20eligible%20for%20this%20%24250%20bonus.%20Also%2C%20internal%20resources%20are%20used%20and%20i%20dont%20want%20them%20to%20be%20wasted%20on%20these%20minor%20things.%20You%20can%20take%20over%20the%20PR.%20Shipping%20the%20fix%20ASAP%20is%20our%20priority.) to @roryabraham that you can take up the PR from here and i consider myself not eligible for bonus bounty and surprisingly its almost the same time(1.5 months) or maybe more the PR hasn't been merged yet.

Let me know if it doesn't make sense. Thanks

**Comment by JmillsExpensify** [comment]
@aneequeahmad Our Contributing.md is pretty clear that if contributors aren't able to merge a PR within 12 business days, the contract is terminated. That's this case. Now I stand by what I said earlier and you're eligible for reporting the issue, but I think that's the extent of it.

**Comment by roryabraham** [comment]
Sorry for the delay in stepping in here @aneequeahmad – I provided more context to the internal team and we've agreed to pay out the original amount of $500 when [your proposal was accepted](https://github.com/Expensify/App/issues/9737#issuecomment-1233175795).

**Comment by JmillsExpensify** [comment]
Cool, thanks for the extra context. That's helpful. Updated the payouts above as a result.

**Comment by aneequeahmad** [comment]
Sounds good, that's so great of you @roryabraham, Thank-you for providing more context 🙇🏼

@JmillsExpensify actually i'm from `startup culture` and because of that i wasn't expert in writing unit tests as i said [here](https://github.com/Expensify/App/issues/9737#:~:text=%40Santhosh%2DSellavel-,I%27m%20working%20on%20unit%20test%20coverage%20and%20simple%20render%20test.%20I%27m%20not%20an%20expert%20in%20writing%20unit%20tests,-.%20Will%20be%20hopefully)`(This PR taught me a lot and gave me confidence about writing unit tests)`. I'm thankful to roryabraham for guiding and helping me about it.

**Comment by JmillsExpensify** [comment]
Awesome, all good. Quick update: [Draft PR](https://github.com/Expensify/App/pull/12168) still a work in progress. When we get to payouts, the summary [is here](https://github.com/Expensify/App/issues/9737#issuecomment-1346500113).

**Comment by JmillsExpensify** [comment]
@roryabraham I wanted to check in on the WIP PR. Do you think you'll be able to circle back to this one, such that we can have Tim and other jump in this week or next?

**Comment by roryabraham** [comment]
The main PR here is ready to review. Created a couple of related follow-up PRs.

**Comment by JmillsExpensify** [comment]
Still waiting for the regression period to pass. In the meantime, we can get the checklist above kicked off.

**Comment by JmillsExpensify** [comment]
Alright, so time to circle back on the payment. Based on the regression from this PR, no bonus is eligible. Further per our Contributing guidelines, we still need to wait until the 19th, or 7 days after the linked fix reaches production. I'm updating the issue title as a result. 

In the meantime, @roryabraham @Santhosh-Sellavel can you handle your portion of the bug zero steps? I'll get a headstart on the regression tests.

**Comment by JmillsExpensify** [comment]
Also while we are at it, based on the regression, that would mean the following payouts now apply. cc @roryabraham for confirmation given the special considerations discussed above. 

 - @aneequeahmad reported this issue. $750
 - @Santhosh-Sellavel handled PR review, so with the regression that comes out to $500. 

Sound right?

**Comment by roryabraham** [comment]
Sounds right 👍🏼

**Comment by JmillsExpensify** [comment]
Offers sent to Contributors. In the meantime, @roryabraham @Santhosh-Sellavel can either of you close out your items on the BZ checklist? I'll get the regression test started.

**Comment by Santhosh-Sellavel** [comment]
I'll think this more like an improvement to the tool tip we can check all off @roryabraham .

**Comment by aneequeahmad** [comment]
@JmillsExpensify Accepted offer, Thanks for mentioning

**Comment by JmillsExpensify** [comment]
You're all set!

**Comment by JmillsExpensify** [comment]
We still need to compensate Santhosh. Keep open until we close that loop.

**Comment by JmillsExpensify** [comment]
Sorry ya'll. I'll close the loop on that test. Monday and late last week got away from me.

**Comment by JmillsExpensify** [comment]
Closed the loop above, so closing this issue.
